presto-main/src/main/java/io/prestosql/operator/WindowOperator.java

```diff
@@ -334,6 +334,16 @@ public boolean isFinished()
         return outputPages.isFinished();
     }
 
+    @Override
+    public ListenableFuture<?> isBlocked()
+    {
+        if (outputPages.isBlocked()) {
+            return outputPages.getBlockedFuture();
+        }
+
+        return NOT_BLOCKED;
```

Comment:
I regret that you cannot simply `return outputPages.getBlockedFuture()`.

add a comment:
```
// We can block e.g. because of self-triggered spill
if (outputPages.isBlocked()) {
```